runner/host/processes.go

```diff
@@ -22,10 +27,17 @@ func (p Process) Run() op.Op {
 		return op.New(p.ID(), processes, op.Fail, err, nil)
 	}
 
-	processInfo := make([]string, 0)
-	for eachProcess := range processes {
-		process := processes[eachProcess]
-		processInfo = append(processInfo, process.Executable())
+	// Maps parent PIDs to child processes
+	var processInfo = make(map[int][]Process)
+
+	for _, process := range processes {
+		proc := Process{
+			Name: process.Executable(),
+			PID:  process.Pid(),
+			PPID: process.PPid(),
+		}
+		// Append to slice of Process under this PPID
+		processInfo[proc.PPID] = append(processInfo[proc.PPID], proc)
```

Comment:
This approach is a little bit between a tree representing parents and childen, and a flat list. What i mean by that, is when we keying and batch processes by their parent they show up multiple times in the readout, sometimes as the key of child processes, and other times as entries.

Handling arbitrary nesting like this gets pretty complicated, so lets instead render them as a flat map of processes with their PIDs, not PPIDs as their keys. For extra credit, a compound key like "{name}.{pid}" (or a pid,name tuple) could make it more readable. But there's edge cases to cover like what if there's a super long filename as the process. Maybe truncate the name to the first and last 5 letters like "pid-proce...name"?

Give it a try but it's ok to cut scope and just use the unique PID keys - the names are still available in the struct!